### DISCUSSION TITLE
How do I access the Session class in_transaction() method from a scoped_session?

### DISCUSSION THREAD

**Discussion by pyper-instacart** [sqlalchemy/sqlalchemy#7965] [Usage Questions] [answered]
It's not included in the proxied methods from the Session class here:

https://github.com/zzzeek/sqlalchemy/blob/fc5c54fcd4d868c2a4c7ac19668d72f506fe821e/lib/sqlalchemy/orm/scoping.py#L75

How can I drill down safely from the scoped_session to the Session object so I can use the in_transaction() method?

**Comment by zzzeek** [comment, marked as answer]
with a scoped_session, you "call" it to get the current session:

```py
the_current_session = my_scoped_session()
```

therefore to call any method, just add the "call", as in 

```py
my_scoped_session().in_transaction()
```

that said, we should add that method to scoped_session.

**Reply by pyper-instacart** [reply]
@zzzeek Oh my god, how did I not I know this!  Thanks!  I was doing dir(my_scoped_session) not dir (my_scoped_session()) and I couldn't see it.  Thanks again.

**Reply by zzzeek** [reply]
the docs for the call thing are in the section at https://docs.sqlalchemy.org/en/14/orm/contextual.html?highlight=scoped_session#unitofwork-contextual .   I'm not sure how often people find that document.

**Reply by karolinepauls** [reply]
TBH, I'm still to see a legitimate use of `__call__` that isn't implementing a simple partial or a cached func.

Having a method called `get_current_session`, `get_proxied_session`, `get_instance` would be much better discoverable in docs and understandable in code.

I feel like `db.session().do_something()` requires an accompanying comment explaining that calling the scoped session returns the current session.

**Reply by zzzeek** [reply]
scoped_session has a method registry() that will give you the current session

   sess = ss.registry()

anyway `__call__` is entirely legitimate and scoped_session() has been the way it is for 16 years, from my POV nobody should be using scoped_session anyway.
